test/integration/runner/runner_test.go

```diff
@@ -0,0 +1,140 @@
+package runner
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"testing"
+)
+
+var timeout = flag.Duration("sub.timeout", 0, "Specify the timeout for each sub test")
+
+func TestIntegration(t *testing.T) {
+	dir := ".."
+	packageName := "github.com/openshift/origin/test/integration"
```

Comment:
Each dir is a binary (has to be in Go).  So you'd just run multiple tests.

As far as how the runner is factored, it's probably going to shift to be arguments, but I want to make sure all this works before refactoring for that.  The runner probably works best as having a number of TestSUITE methods if the dominant call pattern is run all